src_lockfree/htbl/htbl_unsafe.ml

```diff
@@ -0,0 +1,634 @@
+(* Copyright (c) 2023 Vesa Karvonen
+
+   Permission to use, copy, modify, and/or distribute this software for any
+   purpose with or without fee is hereby granted, provided that the above
+   copyright notice and this permission notice appear in all copies.
+
+   THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+   REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+   AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+   INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+   LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+   OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+   PERFORMANCE OF THIS SOFTWARE. *)
+
+open Htbl_utils
+module Atomic = Multicore_magic.Transparent_atomic
+module Atomic_array = Multicore_magic.Atomic_array
```

Comment:
Hmm... The diff between the safe and unsafe versions
```diff
➜  htbl git:(picos_htbl) ✗ diff htbl.ml htbl_unsafe.ml
14d13
< open Htbl_utils
16,28c15,17
< module Atomic_array = struct
<   type 'a t = 'a Atomic.t array
< 
<   let[@inline] at (xs : 'a t) i : 'a Atomic.t = Array.get xs i
<   let[@inline] make n v = Array.init n @@ fun _ -> Atomic.make v
< 
<   external length : 'a array -> int = "%array_length"
< 
<   let unsafe_fenceless_get xs i = Atomic.get xs.(i)
< 
<   let[@inline] unsafe_compare_and_set xs i b a =
<     Atomic.compare_and_set (at xs i) b a
< end
---
> open Htbl_utils
> module Atomic = Multicore_magic.Transparent_atomic
> module Atomic_array = Multicore_magic.Atomic_array
```
seems like something that could probably be handled with a bit of dune build magic.

Reply:
Maybe something along the lines of (this is untested, just looking at [actions](https://dune.readthedocs.io/en/stable/reference/actions/index.html)):

```
(rule 
 (action
  (with-stdout-to htbl.ml
   (progn
    (cat htbl.head_safe.ml)
    (cat htbl.body.ml)))))

(rule 
 (action
  (with-stdout-to htbl_unsafe.ml
   (progn
    (cat htbl.head_unsafe.ml)
    (cat htbl.body.ml)))))
```

Reply:
Works perfectly, thanks !